packages/ripple-address-codec/src/index.ts

```diff
@@ -43,8 +43,10 @@ function encodeXAddress(
   if (tag > MAX_32_BIT_UNSIGNED_INT) {
     throw new Error('Invalid tag')
   }
-  const theTag = tag === false ? 0 : tag
-  const flag = tag === false ? 0 : 1
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- Passing null is a common js mistake
+  const theTag = tag === false || tag == null ? 0 : tag
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- Passing null is a common js mistake
+  const flag = tag === false || tag == null ? 0 : 1
```

Comment:
We can simplify this to:
```suggestion
  const theTag = tag || 0
  const flag = tag ? 1 : 0
```

Reply:
The 1st simplification works, but the 2nd doesn't because if the tag is `0` we need that to end up with `flag = 1` 
("There is a tag, and its value is 0")

Reply:
Oh, then modify the 2nd one to be
```TypeScript
const flag = tag || 0 ? 1 : 0
```